ci-operator/step-registry/cucushift/installer/check/azure/private/lb/cucushift-installer-check-azure-private-lb-commands.sh

```diff
@@ -0,0 +1,125 @@
+else
+    az cloud set --name AzureCloud
+fi
+az login --service-principal -u "${AZURE_AUTH_CLIENT_ID}" -p "${AZURE_AUTH_CLIENT_SECRET}" --tenant "${AZURE_AUTH_TENANT_ID}" --output none
+
+INSTALL_CONFIG="${SHARED_DIR}/install-config.yaml"
+INFRA_ID=$(jq -r .infraID ${SHARED_DIR}/metadata.json)
+RESOURCE_GROUP=$(yq-go r "${INSTALL_CONFIG}" 'platform.azure.resourceGroupName')
+if [[ -z "${RESOURCE_GROUP}" ]]; then
+    RESOURCE_GROUP="${INFRA_ID}-rg"
+fi
+
+publish=$(yq-go r "${INSTALL_CONFIG}" "publish")
+outbound_type=$(yq-go r "${INSTALL_CONFIG}" "platform.azure.outboundType")
+echo -e "--- cluster config --- \npublish: ${publish}\noutbound_type: ${outbound_type}"
+
+if [[ "${publish}" != "Internal" ]]; then
+    echo "This step is applicable on azure private cluster!"
+    exit 1
+fi
+
+critical_check_result=0
+no_critical_check_result=0
+
+#lb check
+lb_list=$(az network lb list -g ${RESOURCE_GROUP} -ojson |jq -r '.[].name')
+echo "INFO: lb list is ${lb_list}"
+lb_internal="${INFRA_ID}-internal"
+if [[ "${lb_list}" =~ ${lb_internal} ]]; then
+    lb_public=$(echo ${lb_list/"${lb_internal}"} | tr -d '\n')
+else
+    echo "ERROR: unable to find internal lb ${lb_internal}!"
+    exit 1
+fi
+
+if [[ "${outbound_type}" == "UserDefinedRouting" ]]; then
+    echo "Cluster outBound type is UserDefinedRouting, this is fully private cluster..."
+    if (( ocp_minor_version >= 11 )); then
+        echo "Check that no public load balancer created on fully private cluster..."
+        [[ -n "${lb_public}" ]] && echo "WARN: Found public load balancer ${lb_public} on fully private cluster!" && no_critical_check_result=1
+    fi
+    # check lb frontended IP config
+    for lb in ${lb_list}; do
+        echo "Check public ip address of Loadbalance ${lb}, should not be created..."
+        ret=0
+        az network lb show -g ${RESOURCE_GROUP} -n ${lb} | jq -r '.frontendIPConfigurations[].publicIPAddress.id' && ret=1
+        [[ ${ret} -eq 0 ]] && echo "ERROR: Found public ip address for load balancer: ${lb}" && critical_check_result=1
+    done
+else
+    echo "Cluster outBound type is loadBalancer, this is private cluster..."
+    echo "Check that outbound rules are created on private cluster..."
+    outbound_rules=$(az network lb show --name ${lb_public} -g ${RESOURCE_GROUP} -ojson | jq -r ".outboundRules[]")
+    if [[ -z "${outbound_rules}" ]]; then
+        echo "ERROR: Not found outbound rules for public load balancer ${lb_public} on private cluster!"
+        critical_check_result=1
+    else
+        echo -e "Found outbound rules for public lb ${lb_public}\n${outbound_rules}"
+    fi
+
+    echo "Check public ip address of Loadbalance ${lb_public}, should be configured..."
+    ret=0
+    az network lb show -g ${RESOURCE_GROUP} -n ${lb_public} | jq -r '.frontendIPConfigurations[].publicIPAddress.id' || ret=1
+    [[ ${ret} -eq 1 ]] && echo "ERROR: Unable to find public ip address for load balancer: ${lb_public}" && critical_check_result=1
```

Comment:
```suggestion
    lb_public_ip=""
    lb_public_ip=$(az network lb show -g ${RESOURCE_GROUP} -n ${lb_public} | jq -r '.frontendIPConfigurations[].publicIPAddress.id')
    [[ -z ${lb_public_ip} ]] && echo "ERROR: Unable to find public ip address for load balancer: ${lb_public}" && critical_check_result=1
```

How about this for easy reading.

Reply:
If we consider az may return `null`, we can `([[ -z ${lb_public_ip} ]] || [[ "${lb_public_ip}" == "null" ]])`